plugins/network-elements/nuage-vsp/src/com/cloud/network/manager/NuageVspManagerImpl.java

```diff
@@ -285,60 +507,341 @@ public boolean deleteNuageVspDevice(DeleteNuageVspDeviceCmd cmd) {
+                        NetworkOfferingServiceMapVO offService =
+                                new NetworkOfferingServiceMapVO(defaultNuageVspSharedSGNetworkOffering.getId(), service, defaultNuageVspSharedSGNetworkOfferingProviders.get(service));
+                        _networkOfferingServiceMapDao.persist(offService);
+                        s_logger.trace("Added service for the NuageVsp network offering: " + offService);
+                    }
+                }
+            }
+        });
+    }
+
+    @DB
+    private void initNuageVspVpcOffering() {
+        //configure default Nuage VSP vpc offering
+        Transaction.execute(new TransactionCallbackNoReturn() {
+            @Override
+            public void doInTransactionWithoutResult(TransactionStatus status) {
+                if (_vpcOffDao.findByUniqueName(nuageVPCOfferingName) == null) {
+                    s_logger.debug("Creating default Nuage VPC offering " + nuageVPCOfferingName);
+
+                    Map<Network.Service, Set<Network.Provider>> svcProviderMap = new HashMap<Network.Service, Set<Network.Provider>>();
+                    Set<Network.Provider> nuageProviders = new HashSet<Network.Provider>();
+                    nuageProviders.add(Network.Provider.NuageVsp);
+                    svcProviderMap.put(Network.Service.Connectivity, nuageProviders);
+                    svcProviderMap.put(Network.Service.Dhcp, nuageProviders);
+                    svcProviderMap.put(Network.Service.StaticNat, nuageProviders);
+                    svcProviderMap.put(Network.Service.SourceNat, nuageProviders);
+                    svcProviderMap.put(Network.Service.NetworkACL, nuageProviders);
+
+                    Set<Network.Provider> userDataProviders = new HashSet<Network.Provider>();
+                    userDataProviders.add(Network.Provider.VPCVirtualRouter);
+                    svcProviderMap.put(Network.Service.UserData, userDataProviders);
+
+                    createVpcOffering(nuageVPCOfferingName, nuageVPCOfferingDisplayText, svcProviderMap, true, VpcOffering.State.Enabled, null);
+                }
+            }
+        });
+    }
+
+    @DB
+    protected VpcOffering createVpcOffering(final String name, final String displayText, final Map<Network.Service, Set<Network.Provider>> svcProviderMap, final boolean isDefault,
+                                            final VpcOffering.State state, final Long serviceOfferingId) {
+        return Transaction.execute(new TransactionCallback<VpcOffering>() {
+            @Override
+            public VpcOffering doInTransaction(TransactionStatus status) {
+                // create vpc offering object
+                VpcOfferingVO offering = new VpcOfferingVO(name, displayText, isDefault, serviceOfferingId, false, false);
+
+                if (state != null) {
+                    offering.setState(state);
+                }
+                s_logger.debug("Adding vpc offering " + offering);
```

Comment:
Wrap in a if (s_logger.isDebugEnabled()) { ... } block to avoid unnecessary pressure on the string pool when debug logging is not enabled.